pkg/registrar/map_functions.go

```diff
@@ -133,6 +138,66 @@ func (mapper *Mapper) RunTemplateToRunnableRequests(object client.Object) []reco
 	return requests
 }
 
+// addGVK fulfills the 'GVK of an object returned from the APIServer
+// https://github.com/kubernetes-sigs/controller-runtime/issues/1517#issuecomment-844703142
+func (mapper *Mapper) addGVK(obj client.Object) error {
+	gvks, unversioned, err := mapper.Client.Scheme().ObjectKinds(obj)
+	if err != nil {
+		return fmt.Errorf("missing apiVersion or kind: %s err: %w", obj.GetName(), err)
+	}
+
+	if unversioned {
+		return fmt.Errorf("unversioned object: %s", obj.GetName())
+	}
+
+	if len(gvks) != 1 {
+		return fmt.Errorf("unexpected GVK count: %s", obj.GetName())
+	}
+
+	obj.GetObjectKind().SetGroupVersionKind(gvks[0])
+	return nil
+}
+
+func (mapper *Mapper) TemplateToSupplyChainRequests(template client.Object) []reconcile.Request {
+
+	templateName := template.GetName()
+
+	err := mapper.addGVK(template)
+	if err != nil {
+		mapper.Logger.Error(err, fmt.Sprintf("could not get GVK for template: %s", templateName))
+		return nil
+	}
+
+	list := &v1alpha1.ClusterSupplyChainList{}
+
+	err = mapper.Client.List(
+		context.TODO(),
+		list,
+	)
+
+	if err != nil {
+		mapper.Logger.Error(err, "template to supply chain requests")
```

Comment:
wdyt of

```suggestion
		mapper.Logger.Error(err, "list cluster supplychain")
```

?

`template to supply chain requests` would be describing the whole function, rather than just the problem that actually occurred

Reply:
new commit to fix